### ISSUE TITLE
Method not found: 'Microsoft.Practices.Unity.IUnityContainer ...ContainerManager.GetConfiguredContainer()

### ISSUE THREAD

**Issue by sparra1000** [unitycontainer/container#139]
I am getting this error each time I change something in the web.config on a production azure asp.net mvc web app.  The application runs fine wjhen it is published from visual studio but after it recompiles on the azure web service this error always appears:
 
Server Error in '/' Application.
Method not found: 'Microsoft.Practices.Unity.IUnityContainer MyProject.Core.ContainerManager.GetConfiguredContainer()'.
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.MissingMethodException: Method not found: 'Microsoft.Practices.Unity.IUnityContainer MyProject.Core.ContainerManager.GetConfiguredContainer()'.

Source Error:

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace:

[MissingMethodException: Method not found: 'Microsoft.Practices.Unity.IUnityContainer MyProject.Core.ContainerManager.GetConfiguredContainer()'.]
   MyProject.Web.App_Start.UnityWebActivator.Start() +0

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor) +0
   System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments) +87
   System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) +101
   WebActivatorEx.BaseActivationMethodAttribute.InvokeMethod() +73
   WebActivatorEx.ActivationManager.RunActivationMethods(Boolean designerMode) +350
   WebActivatorEx.ActivationManager.Run() +78

[InvalidOperationException: The pre-application start initialization method Run on type WebActivatorEx.ActivationManager threw an exception with the following error message: Exception has been thrown by the target of an invocation..]
   System.Web.Compilation.BuildManager.InvokePreStartInitMethodsCore(ICollection`1 methods, Func`1 setHostingEnvironmentCultures) +615
   System.Web.Compilation.BuildManager.InvokePreStartInitMethods(ICollection`1 methods) +141
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +102
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +157
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +549

[HttpException (0x80004005): The pre-application start initialization method Run on type WebActivatorEx.ActivationManager threw an exception with the following error message: Exception has been thrown by the target of an invocation..]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +10075124
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +95
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.7.3282.0

**Comment by ENikS** [comment]
You are running very old version. It is no longer supported.
Just by looking at exception though, I could guess that something is wrong with your MyProject.Core.ContainerManager

**Comment by sparra1000** [comment]
I'm running 3.5.0.0.  Are there significant breaking changes to upgrade to the latest version from 3.5.00?

**Comment by ENikS** [comment]
Yes. It changed somewhat in last 8 years

**Comment by ENikS** [comment]
Actually, if you are willing to document all the issues you had to deal with during upgrade and create little tutorial for others to follow I would be willing to help you in your task. [This wiki](https://github.com/unitycontainer/unity/wiki) has very basic info on the differences in later versions but I'd like to create more detailed tutorial.

**Comment by sparra1000** [comment]
I will do that.  I don't have much time right now but will eventually get to it since it causes a lot of issues with the application and it is very unstable because of this.  As I start I'll let you know, is there a way to communicate besides these messages?

**Comment by ENikS** [comment]
If we discuss it here others could follow with all the issues you found

**Comment by sparra1000** [comment]
I'lm starting to upgrade from 3.10 to 4.0.1.  I want to do this first then I'll go to the most recent version.  I feel this is the smallest upgrade to get it to work.  I'll document what I did.  To start, I did a nuget Install-Package Unity.Mvc -Version 4.0.1 in all relevant projects in the solution.  I let the package manager overwrite UnityMvcActivator.cs and UnityConfig.cs.  Then I added back my RegisterTypes into UnityConfig.

I'm getting this error:

    System.Exception
  HResult=0x80131500
  Message=Global.asax -Resolution of the dependency failed, type = "Repository.Pattern.DataContext.IDataContextAsync", name = "(none)".
Exception occurred while: while resolving.
Exception is: InvalidOperationException - The current type, Repository.Pattern.DataContext.IDataContextAsync, is an interface and cannot be constructed. Are you missing a type mapping?
-----------------------------------------------
At the time of the exception, the container was:

  Resolving Repository.Pattern.DataContext.IDataContextAsync,(none)

  Source=MyProject.Web
  StackTrace:
   at MyProject.Web.MvcApplication.Application_BeginRequest(Object sender, EventArgs e) in D:\master\src\MyProject.Web\Global.asax.cs:line 129
